Makefile

```diff
@@ -646,7 +646,7 @@ go-test: $(SETUP_ENVTEST) ## Run go tests.
 
 .PHONY: test-cover
 test-cover: TEST_ARGS+= -coverprofile coverage.out
-test-cover: go-test-race ## Run tests with code coverage and generate reports.
```

Comment:
this stops running the race-detector on PRs (essentially reverting #2646), do we want to find a way to do both? maybe add lint to the existing target?

Reply:
Sorry for commenting uninvited, I was looking at this PR the same time. 
With this change, looks like pre-requisite `test` for the target `test-cover` invokes `go-test-race` while running the target [`test`](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/64ac8815dacc6c086eb31af01b33d9db2be564ea/Makefile#L637). 
So, looks like we are good to go.

Reply:
This PR makes it seem as if target-specific variables aren't working in the CI pipeline. See [this comment](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2646#pullrequestreview-1209063664). Perhaps #2646 was where this broke?

Reply:
@CecileRobertMichon -- The `test` target has already `go-test-race` as dependency. 
Also -- I validated it by running locally that race detector is enabled. This is what I get in my log : 
```
KUBEBUILDER_ASSETS="/Users/kuashutosh/Library/Application Support/io.kubebuilder.envtest/k8s/1.20.2-darwin-amd64" go test ./... -coverprofile coverage.out -race
```
Let me know if I am still missing something.

Reply:
See line 636-641 on this file.